### PR TITLE
quoting config not working with 1.8 

### LINKED ISSUE #1074: [Bug] `quoting` config not working with 1.8
### Is this a new bug in dbt-snowflake?

- [X] I believe this is a new bug in dbt-snowflake
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

The `quoting` config doesn't quote the database anymore in 1.8 unlike it did pre-1.8.

### Expected Behavior

The `quoting` configs should quote the database like it did pre-1.8.

### Steps To Reproduce

1. Create a database that starts with a number:
```sql
create database "1_JYEO";
```

2. Use the database in the dbt project with `quoting.database = true`:

```yml
# dbt_project.yml
name: my_dbt_project
profile: all
config-version: 2
version: "1.0.0"

models:
 my_dbt_project:
    +materialized: table
    +database: 1_JYEO

quoting:
  database: true 
```

```sql
-- models/foo.sql
select 1 id
```

3. Run on 1.8:

```sh
$ dbt --debug run

01:07:44  Sending event: {'category': 'dbt', 'action': 'invocation', 'label': 'start', 'context': [<snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x1046b1510>, <snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x1046efa90>, <snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x1046f7fd0>]}
01:07:44  Running with dbt=1.8.1
01:07:44  running dbt with arguments {'printer_width': '80', 'indirect_selection': 'eager', 'write_json': 'True', 'log_cache_events': 'False', 'partial_parse': 'True', 'cache_selected_only': 'False', 'warn_error': 'None', 'fail_fast': 'False', 'profiles_dir': '/Users/jeremy/.dbt', 'log_path': '/Users/jeremy/git/dbt-basic/logs', 'debug': 'True', 'version_check': 'True', 'use_colors': 'True', 'use_experimental_parser': 'False', 'no_print': 'None', 'quiet': 'False', 'warn_error_options': 'WarnErrorOptions(include=[], exclude=[])', 'introspect': 'True', 'log_format': 'default', 'invocation_command': 'dbt --debug run', 'target_path': 'None', 'static_parser': 'True', 'send_anonymous_usage_stats': 'True'}
01:07:45  Sending event: {'category': 'dbt', 'action': 'project_id', 'label': '7e16f034-b8c9-429a-85e9-e728b206eccf', 'context': [<snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x13518a5d0>]}
01:07:45  Sending event: {'category': 'dbt', 'action': 'adapter_info', 'label': '7e16f034-b8c9-429a-85e9-e728b206eccf', 'context': [<snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x104142590>]}
01:07:45  Registered adapter: snowflake=1.8.2
01:07:45  checksum: ebc509fb14e151ea268e89b60fd5abf912ed99129f87516462718fa2f14f4838, vars: {}, profile: , target: , version: 1.8.1
01:07:45  Unable to do partial parsing because of a version mismatch
01:07:45  Sending event: {'category': 'dbt', 'action': 'partial_parser', 'label': '7e16f034-b8c9-429a-85e9-e728b206eccf', 'context': [<snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x10335f790>]}
01:07:46  Sending event: {'category': 'dbt', 'action': 'load_project', 'label': '7e16f034-b8c9-429a-85e9-e728b206eccf', 'context': [<snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x1353b2750>]}
01:07:46  Sending event: {'category': 'dbt', 'action': 'resource_counts', 'label': '7e16f034-b8c9-429a-85e9-e728b206eccf', 'context': [<snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x136280b90>]}
01:07:46  Found 1 model, 558 macros
01:07:46  Sending event: {'category': 'dbt', 'action': 'runnable_timing', 'label': '7e16f034-b8c9-429a-85e9-e728b206eccf', 'context': [<snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x136451110>]}
01:07:46  
01:07:46  Acquiring new snowflake connection 'master'
01:07:46  Acquiring new snowflake connection 'list_1_JYEO'
01:07:46  Using snowflake connection "list_1_JYEO"
01:07:46  On list_1_JYEO: /* {"app": "dbt", "dbt_version": "1.8.1", "profile_name": "all", "target_name": "sf", "connection_name": "list_1_JYEO"} */
show terse schemas in database "1_JYEO"
    limit 10000
01:07:46  Opening a new connection, currently in state init
01:07:49  SQL status: SUCCESS 3 in 3.0 seconds
01:07:49  On list_1_JYEO: Close
01:07:49  Acquiring new snowflake connection 'list_1_JYEO_dbt_jyeo'
01:07:49  Using snowflake connection "list_1_JYEO_dbt_jyeo"
01:07:49  On list_1_JYEO_dbt_jyeo: /* {"app": "dbt", "dbt_version": "1.8.1", "profile_name": "all", "target_name": "sf", "connection_name": "list_1_JYEO_dbt_jyeo"} */
show objects in 1_JYEO.dbt_jyeo limit 10000
01:07:49  Opening a new connection, currently in state init
01:07:51  Snowflake adapter: Snowflake query id: 01b4d6e3-0804-b779-000d-3783366a862a
01:07:51  Snowflake adapter: Snowflake error: 001003 (42000): SQL compilation error:
syntax error line 1 at position 16 unexpected '1'.
01:07:51  Snowflake adapter: Error running SQL: macro list_relations_without_caching
01:07:51  Snowflake adapter: Rolling back transaction.
01:07:51  On list_1_JYEO_dbt_jyeo: Close
01:07:51  Connection 'master' was properly closed.
01:07:51  Connection 'list_1_JYEO' was properly closed.
01:07:51  Connection 'list_1_JYEO_dbt_jyeo' was properly closed.
01:07:51  
01:07:51  Finished running  in 0 hours 0 minutes and 5.47 seconds (5.47s).
01:07:51  Encountered an error:
Database Error
  001003 (42000): SQL compilation error:
  syntax error line 1 at position 16 unexpected '1'.
01:07:51  Resource report: {"command_name": "run", "command_wall_clock_time": 7.0150514, "process_user_time": 3.133745, "process_kernel_time": 1.425414, "process_mem_max_rss": "202407936", "command_success": false, "process_in_blocks": "0", "process_out_blocks": "0"}
01:07:51  Command `dbt run` failed at 13:07:51.939918 after 7.02 seconds
01:07:51  Sending event: {'category': 'dbt', 'action': 'invocation', 'label': 'end', 'context': [<snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x104717f10>, <snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x104717e90>, <snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x100afc310>]}
01:07:51  Flushing usage events
```

> `show objects in 1_JYEO.dbt_jyeo limit 10000` invalid.

4. Revert to 1.7 and then run again:

```sh
$ dbt --debug run

01:08:36  On list_1_JYEO: /* {"app": "dbt", "dbt_version": "1.7.14", "profile_name": "all", "target_name": "sf", "connection_name": "list_1_JYEO"} */
show terse schemas in database "1_JYEO"
    limit 10000
01:08:36  Opening a new connection, currently in state init
01:08:37  SQL status: SUCCESS 3 in 2.0 seconds
01:08:37  On list_1_JYEO: Close
01:08:38  Re-using an available connection from the pool (formerly list_1_JYEO, now list_1_JYEO_dbt_jyeo)
01:08:38  Using snowflake connection "list_1_JYEO_dbt_jyeo"
01:08:38  On list_1_JYEO_dbt_jyeo: /* {"app": "dbt", "dbt_version": "1.7.14", "profile_name": "all", "target_name": "sf", "connection_name": "list_1_JYEO_dbt_jyeo"} */
show terse objects in "1_JYEO".dbt_jyeo limit 10000
01:08:38  Opening a new connection, currently in state closed
01:08:40  SQL status: SUCCESS 2 in 2.0 seconds
01:08:40  On list_1_JYEO_dbt_jyeo: Close
01:08:40  Sending event: {'category': 'dbt', 'action': 'runnable_timing', 'label': '2678a36e-2a7c-4b6c-9c36-a03733b3b4ac', 'context': [<snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x154929ad0>]}
01:08:40  Concurrency: 1 threads (target='sf')
01:08:40  
01:08:40  Began running node model.my_dbt_project.foo
01:08:40  1 of 1 START sql table model 1_JYEO.dbt_jyeo.foo ............................... [RUN]
01:08:40  Re-using an available connection from the pool (formerly list_1_JYEO_dbt_jyeo, now model.my_dbt_project.foo)
01:08:40  Began compiling node model.my_dbt_project.foo
01:08:40  Writing injected SQL for node "model.my_dbt_project.foo"
01:08:40  Timing info for model.my_dbt_project.foo (compile): 13:08:40.768038 => 13:08:40.773299
01:08:40  Began executing node model.my_dbt_project.foo
01:08:40  Writing runtime sql for node "model.my_dbt_project.foo"
01:08:40  Using snowflake connection "model.my_dbt_project.foo"
01:08:40  On model.my_dbt_project.foo: /* {"app": "dbt", "dbt_version": "1.7.14", "profile_name": "all", "target_name": "sf", "node_id": "model.my_dbt_project.foo"} */
create or replace transient table "1_JYEO".dbt_jyeo.foo
         as
        (select 1 id
        );
01:08:40  Opening a new connection, currently in state closed
01:08:42  SQL status: SUCCESS 1 in 2.0 seconds
01:08:42  Timing info for model.my_dbt_project.foo (execute): 13:08:40.773821 => 13:08:42.848314
01:08:42  On model.my_dbt_project.foo: Close
01:08:43  Sending event: {'category': 'dbt', 'action': 'run_model', 'label': '2678a36e-2a7c-4b6c-9c36-a03733b3b4ac', 'context': [<snowplow_tracker.self_describing_json.SelfDescribingJson object at 0x154ada510>]}
01:08:43  1 of 1 OK created sql table model 1_JYEO.dbt_jyeo.foo .......................... [SUCCESS 1 in 2.71s]
```

> `show terse objects in "1_JYEO".dbt_jyeo limit 10000` valid.

### Relevant log output

_No response_

### Environment

```markdown
- OS: macOS
- Python: 3.11
- dbt-core: 1.8.2
- dbt-snowflake: 1.8.2
```

### Additional Context

I don't think schema quoting is working either.

1.8:
https://github.com/dbt-labs/dbt-snowflake/blob/cc1388885c596a251489e65d6a8f97a0985a000e/dbt/include/snowflake/macros/adapters.sql#L127

1.7:
https://github.com/dbt-labs/dbt-snowflake/blob/f1fc9480a8dcf9b28c5cd9b070a044cf63714894/dbt/include/snowflake/macros/adapters.sql#L127

### PR DESCRIPTION
**Pull request by McKnight-42**
resolves #1074 
[docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose) dbt-labs/docs.getdbt.com/#

### Problem
During decoupling work we updated some naming conventions in some macros that seem to go around quotiing_policy, reverting to put us back in good standing.

### Solution
revert to using `schema_relation` where able in the `adapters.sql` if schema_relation is a string, otherwise use `schema_relation.include(indentifier=False)` 
- interactions in test seem to suggest both version are needed see `tests/functional/adapter/list_relations_tests/test_pagination.py`

update tests in `tests/functional/adapter/list_relations_tests/test_pagination.py` to use a `SnowflakeRelation` object instead of a `dict`

add a new test based of the edgecase from issue.

### Todo:
- [x] Make change in macro
- [x] Update current tests
- [x] make new test
- [ ] **backport to 1.8.latest** after merge

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [ ] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc) or this PR has already received feedback and approval from Product or DX

### CODE REVIEW

**Comment by mikealfare** [comment]
We do this in two spots, so this [line](https://github.com/dbt-labs/dbt-snowflake/blob/0fe48e65c28f32dd3958927ca110dd800fe1701a/dbt/include/snowflake/macros/adapters.sql#L127) would also need updating.

File: tests/functional/adapter/list_relations_tests/test_pagination.py [hunk 1]
```diff
@@ -122,8 +121,8 @@ def test__snowflake__list_relations_without_caching_termination(self, project):
         schemas = project.created_schemas
 
         for schema in schemas:
-            schema_relation = {"database": database, "schema": schema}
-            kwargs = {"schema_relation": schema_relation}
+            schema_relation = SnowflakeRelation.create(database=database, schema=schema)
+            kwargs = {"schema_relation": schema_relation.render()}
```
Comment (mikealfare) [inline tests/functional/adapter/list_relations_tests/test_pagination.py:125]: Is there a reason to render here? Are we testing if the macro can take a relation object or a string (or both I suppose)?

File: tests/functional/adapter/list_relations_tests/test_special_characters.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+{{ config(schema = '1_contains_special*character$') }}
+select 1 as id
+"""
+
+
+class TestSpecialCharactersInSchema:
+
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {"quoting": {"schema": True, "identifier": True}}
```
Comment (mikealfare) [inline tests/functional/adapter/list_relations_tests/test_special_characters.py:16]: Per a private discussion, I agree that we should not need to specify `identifier` here. I think what's happening is that you're naming the table `table` (from `"table.sql"` below) and that's a reserved word. Try something like `my_table` and see if you can remove `identifier` here.

File: tests/functional/adapter/list_relations_tests/test_pagination.py (hunk 1 above)
Comment (colin-rogers-dbt) [reply tests/functional/adapter/list_relations_tests/test_pagination.py:125]: you can't pass a proper python object to run-operation AFAIK so this is an easy workaround

File: tests/functional/adapter/list_relations_tests/test_special_characters.py (hunk 2 above)
Comment (McKnight-42) [reply tests/functional/adapter/list_relations_tests/test_special_characters.py]: updated to the generic my_model format

**Review by colin-rogers-dbt — APPROVED** [review]

**Pull request merged**